modules/security/src/test/java/io/oasp/module/security/common/impl/accesscontrol/AccessControlConfigTest.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Test of {@link AccessControlConfig}.
+ */
+public class AccessControlConfigTest extends ModuleTest {
+
+  @Test
+  // @RolesAllowed only used to ensure that the constant can be referenced here
+  @RolesAllowed(AccessControlConfigSimple.PERMISSION_FIND_TABLE)
+  public void testSimple() {
+
+    // given + when
+    AccessControlConfigSimple config = new AccessControlConfigSimple();
+    AccessControlGroup chief = (AccessControlGroup) config.getAccessControl(AccessControlConfigSimple.GROUP_CHIEF);
+
+    // then
+    assertThat(chief).isNotNull();
+    assertThat(chief.getId()).isEqualTo(AccessControlConfigSimple.GROUP_CHIEF);
+    assertThat(flatten(chief.getPermissions())).containsExactly(AccessControlConfigSimple.PERMISSION_SAVE_OFFER,
+        AccessControlConfigSimple.PERMISSION_SAVE_PRODUCT, AccessControlConfigSimple.PERMISSION_SAVE_STAFF_MEMBER,
+        AccessControlConfigSimple.PERMISSION_DELETE_OFFER, AccessControlConfigSimple.PERMISSION_DELETE_PRODUCT,
+        AccessControlConfigSimple.PERMISSION_DELETE_STAFF_MEMBER,
+        AccessControlConfigSimple.PERMISSION_DELETE_ORDER_POSITION, AccessControlConfigSimple.PERMISSION_DELETE_TABLE);
+    AccessControlGroup waiter = getSingleInherit(chief);
+    assertThat(waiter).isNotNull();
+    assertThat(waiter.getId()).isEqualTo(AccessControlConfigSimple.GROUP_WAITER);
+    assertThat(flatten(waiter.getPermissions())).containsExactly(AccessControlConfigSimple.PERMISSION_SAVE_TABLE);
+    AccessControlGroup barkeeper = getSingleInherit(waiter);
+    assertThat(barkeeper).isNotNull();
+    assertThat(barkeeper.getId()).isEqualTo(AccessControlConfigSimple.GROUP_BARKEEPER);
+    assertThat(flatten(barkeeper.getPermissions())).containsExactly(AccessControlConfigSimple.PERMISSION_FIND_BILL,
+        AccessControlConfigSimple.PERMISSION_SAVE_BILL, AccessControlConfigSimple.PERMISSION_DELETE_BILL,
+        AccessControlConfigSimple.PERMISSION_DELETE_ORDER);
+    AccessControlGroup cook = getSingleInherit(barkeeper);
+    assertThat(cook).isNotNull();
+    assertThat(cook.getId()).isEqualTo(AccessControlConfigSimple.GROUP_COOK);
+    assertThat(flatten(cook.getPermissions())).containsExactly(AccessControlConfigSimple.PERMISSION_FIND_ORDER,
+        AccessControlConfigSimple.PERMISSION_SAVE_ORDER, AccessControlConfigSimple.PERMISSION_FIND_ORDER_POSITION,
+        AccessControlConfigSimple.PERMISSION_SAVE_ORDER_POSITION);
+    AccessControlGroup readMasterData = getSingleInherit(cook);
+    assertThat(readMasterData).isNotNull();
+    assertThat(readMasterData.getId()).isEqualTo(AccessControlConfigSimple.GROUP_READ_MASTER_DATA);
+    assertThat(flatten(readMasterData.getPermissions())).containsExactly(
+        AccessControlConfigSimple.PERMISSION_FIND_OFFER, AccessControlConfigSimple.PERMISSION_FIND_PRODUCT,
+        AccessControlConfigSimple.PERMISSION_FIND_STAFF_MEMBER, AccessControlConfigSimple.PERMISSION_FIND_TABLE);
+    assertThat(readMasterData.getInherits()).isEmpty();
+
+    // and when
```

Comment:
"and when" Anti-pattern. Please seperate tests to let them as of just one reason (in theory).

Reply:
Sometimes being a little more pragmatic that the ideal theory is fine for me. I do not see any added value by separating this but I will do as it is faster done than further discussion...

Reply:
It is always easier to be someone's hero, than later teach him why you did it wrong and he should not.